### DISCUSSION TITLE
Passing a file type for api request

### DISCUSSION THREAD

**Discussion by shubhambajad** [apideck-libraries/portman#359] [Q&A] [answered]
Hi team,

Is there any way in portman using which we can pass the file which we want to upload via api request?

**Comment by thim81** [comment]
hi @shubhambajad 

I havent tried this myself but a combination of parameters, could do the trick

`portman --includeTests false --syncPostman  --localPostman ./your-folder/your.postman.json`

--localPostman ./your-folder/your.postman.json => instructs Portman to skip the OAS to Postman conversion, and use a valid Postman collection JSON file
--includeTests false => instructs Portman to skip any modifications to the Postman collection just loaded
--syncPostman => instructs Portman to upload the loaded Postman collection

Could you report back if it worked?

**Reply by shubhambajad** [reply]
Hi @thim81 

Actually the thing is I have one api request in which I need to pass a file as input like given below

<img width="643" alt="fileUpload" src="https://user-images.githubusercontent.com/95845909/176378586-0f1d6721-a939-4347-bf0f-052f8a192ff3.png">

And I need to pass this at a run time using portman config so Is there any way like from pre-requisite script or something else?

**Comment by thim81** [comment, marked as answer]
Hmm I'm not sure if we would support "file_path" overwrites. 

Can you set this a file and if you export the Postman collection, how would that part look in JSON?

**Reply by shubhambajad** [reply]
Hi @thim81 

I tried one solution like I have used the overwrite property and passed the file path from pre-requisite script where I have mentioned the form data and it worked.

```js
const body = {
        "mode": "formdata",
        "formdata": [{
                "key": "file_path",
                "type": "file",
                "src": "Testing1.json"
            }
        };
pm.request.body.update(body);
```

We can pass relative path by maintaining the file in postman or newman directory or we can pass relative path as well

Thank you for your support.

**Reply by thim81** [reply]
@shubhambajad Thanks for sharing the solutions, so that other Portman users can re-use it.

**Comment by thim81** [comment]
Related to this topic, be sure to have a look at the Portman config technique shared by @Lars-Brebels-92
https://github.com/apideck-libraries/portman/issues/387#issuecomment-1702471592
